### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by josuerushanika**
**For this  milestone of my portfolio website, you I save the form data in the local storage of the browser. That way when the user reloads the page the data they filled out in the form will be preserved:**

_I implement the following interactions:
When the user changes the content of any input field, the data is saved to the local storage.
When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
I use the following data model:
I create a single JavaScript object with all the data from the entire form and save it in local storage. Do not create one local storage entry per input field._

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -95,14 +95,41 @@ for (let index = 0; index < extendProjectBtns.length; index++) {
...
+    Name: name.value,
+    Email: email.value,
+    Message: feedback.value,
+  };
+  // store the object
+  localStorage.setItem('visitor', JSON.stringify(visitor));
+}
+
+// put focus out fonction in function visitor data
+// [name, email, feedback].forEach((input) => input.addEventListener('focusout', visitorData));
```
Comment (danmainah) [inline main.js]: - [ ] @team its not good practise to have commented code in your codebase, kindly remove the commented code.

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ After you're finished please remove all the comments and instructions!
 
 </div>
 
-<!-- TABLE OF CONTENTS -->
+<!-- TABLE OF CONTENTS -->....
```
Comment (danmainah) [inline README.md]: - [ ] kindly ensure your MIT license link is working.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Team , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: main.js (hunk 1 above)
Comment (josuerushanika) [reply main.js]: Thanks it is fixed

File: README.md (hunk 2 above)
Comment (josuerushanika) [reply README.md]: Thanks it is fixed

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ You removed the highlighted commented code

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
- [ ] Your MIT license is not yet fixed as the previous reviewer pointed out, kindly ensure it is fixed

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- I suggest you remove every other commented code in your codebase as this makes your code cleaner and more professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by pinkmoon25 — APPROVED** [review]
Hi @josuerushanika , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
linter checks :heavy_check_mark: 
professional readme :heavy_check_mark: 
preserve data in local storage :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by josuerushanika** [comment]
@pinkmoon25  Thanks It is fixed

**Pull request merged**